### ISSUE TITLE
DAILY UPDATE AT DAY-END

### ISSUE THREAD

**Issue by dhruvin-restalgo** [RestAlgo/TradingApplication#202]
Enter Daily Updates with basic bullet points.

e.g.

-  Developed : user management in Admin login
-  Developing in progress : risk form design, limit management form etc.
-  Read book on sql
-  Took refence for designing from xyz place.

**Comment by Kartik-RestAlgo** [comment]
- Execl Formulas Testing in Live BroadCast
- Completing changes of Admine Module of frmNormalBuy and frmNormalSell
- Removing Extra commented Codes from frmCreateWatch, frmOrderbookList, frmNormalBuy and frmNormalSell
- Removing Autofill Client Code Logic from Admin module.

**Comment by Kartik-RestAlgo** [comment]
- Check Execl Formulas
- Mergeing Code To PratikBhai
- Correcting Market Depth in market Watch
- **New Task** Fill Failed order window and save it by Local Storage.
- Testing text file compress into zip (for reduced size)

**Comment by Kartik-RestAlgo** [comment]
**Development Completed** - Worked on Excel with Lokesh Sir and Vaibhav Sir.
**Development in Progress** - Handled and saved failed messages from the engine to admin.

**Comment by Kartik-RestAlgo** [comment]
**Development in Progress** -Admin should login only once with single login id
**Testing** - Failed order Save in file at every minute

**Comment by Kartik-RestAlgo** [comment]
17-06-2024
===============
**Development in Progress**: Admin should log in only once with a single login ID.
**Development in Progress**: Code separation is underway with Pratik Bhai in the engine.

**Comment by Kartik-RestAlgo** [comment]
19-06-2024
========

**Development in Done**: Created a Common Class for all models on a User-Wise basis.
**Development in Done**: Admin should log in only once with a single login ID.
**Development in Progress**: Data separator and Get/Set Data methods are pending.

**Comment by Kartik-RestAlgo** [comment]
20-06-2024
==========
**Development in Progress:** Save UserWise ConfigSetting Working with Pratik Sir

**Comment by Kartik-RestAlgo** [comment]
21-06-2024
============
**Development Done**: Save UserWise ConfigSetting (Read/Write) Working with Pratik Sir
**Testing** :  Admin should log in only once with a single login ID.

**Comment by Kartik-RestAlgo** [comment]
22-06-2024
================
**Testing** : Admin Module and UserConfig

**Comment by Kartik-RestAlgo** [comment]
01-07-2024
============
**Testing** : Server Client in Local - if 1 user is connected than another user try to login with same Username than 1st user will disconnected
**Client Form BtnSend_Click** - line 72-83, condition add for send username
**Client Form LogoutResponse** - method add line 106-119 to close Form
**Client Form ClientM** - LogoutResponse method add line 26-30 to close Form
**Server Form ServerM LoginRequest** - Correction as per testing line 18-45, line 56-60
**ServerPractice Form** - testing new Logic line 28-114

**Comment by Kartik-RestAlgo** [comment]
02-07-2024
============
**New Project** - Zip unzip file and save with new name always
**Server-Client** - R&D in Project
**Learn** - Learn About NoeMax

**Comment by Kartik-RestAlgo** [comment]
03-07-2024
============
**New Project** - Async And Await Session and Project Line 1-143
**Learn** - Learn About Ienumrable and Ienumrator

**Comment by Kartik-RestAlgo** [comment]
04-07-2024
===========
**New Project** - LoopsDemo, For, Foreach, While, Parallel For, Parallel, Foreach
     **Model** - line no : 1 to 22
     **DataProvider Class** - line no : 1 to 171
     **Appglobal** - line no : 1 to 14
     **Form Code** - line no : 1 to 203
     **Form Design** - line no : Designing
**Learn** - Learn About Aysnc and await from Tm Cory Video
**Code Merging** - Task no 10 : NSEFO validation code merge to Virag Bhai

**Comment by Kartik-RestAlgo** [comment]
05-07-2024
============
**Code Merging** - Task no 17 : IndexAverage Form Merge With Viragbhai

**Comment by Kartik-RestAlgo** [comment]
06-07-2024
============
**Modify** - 2 class UserConfigManager, Appglobal for Properties.

**Comment by Kartik-RestAlgo** [comment]
08-07-2024
========
**Modify** -  UserConfigManager, Appglobal for Properties.
**Developement In Process** - Change Json Format of Use Config File, and apply new Properties

**Comment by Kartik-RestAlgo** [comment]
09-07-2024
================

**Merge** - user Config Class and Related Files And Code
**Developement In Process** - IndexAverage Form

**Comment by Kartik-RestAlgo** [comment]
10-07-2024
===============
**Developement Done** - Find CEToken And PEToken from strike price IndexAverage Form 
**Developement Done** - Order History will save in bin file in Admine Module

**Comment by Kartik-RestAlgo** [comment]
11-07-2024
=================
**Developement In Process** - Filtering in NetPosition using LINQ

**Comment by Kartik-RestAlgo** [comment]
12-07-2024
=================
**Developement In Process** - Filtering in NetPosition using LINQ Updating.
**Developement In Done** - Handling Open NetPosition in Trading system

**Comment by Kartik-RestAlgo** [comment]
13-07-2024
===========
Developement In Done- Filtering in NetPosition using LINQ Updating.

**Comment by Kartik-RestAlgo** [comment]
15-07-2024
========================================
**Developement In Process**- Filtering in NetPosition using LINQ Updating, for M2M and unreliazeM2M with Pratikbhai

**Comment by Kartik-RestAlgo** [comment]
16-07-2024
===========
**Developement Done**- Filtering in NetPosition using LINQ Updating, for M2M and unreliazeM2M with Pratikbhai, Merging Pending

**Comment by Kartik-RestAlgo** [comment]
17-07-2024
===========================

**Testing**- Filtering in NetPosition using LINQ Updating, for M2M and unreliazeM2M with Pratikbhai, 
**Merging** - finding CEToken And PEToken new method merging with viragbhai in Index average form

**Comment by Kartik-RestAlgo** [comment]
18-07-2024
-------------

**Testing**- Filtering in NetPosition using LINQ Updating with Pratikbhai,

**Comment by Kartik-RestAlgo** [comment]
19-07-2024
----------------------------
**Testing** - Filtering in NetPosition using LINQ Updating with Pratikbhai,
**Testing** - FrmIndexAverage testing
**Merge** - LINQ in NetPosition
**Merge** - Only Branch can Login in Admin Module
**Testing & Merge** - Linq in netposition in Admine Module
**Developement In Process** - OTP generate and testing

**Comment by Kartik-RestAlgo** [comment]
20-07-2024
-------------------------
**Developement In ProcessLocal** - Trade: Dummy Trade, not actual trade.

**Comment by Kartik-RestAlgo** [comment]
22-07-2024
=====================

**Developement In Done** - Local Trade: Dummy Trade, not actual trade. localBuy form Developement done and Now on Hold
**Testing** - Linq in netposition in Admine Module with PratikBhai

**Comment by Kartik-RestAlgo** [comment]
23-07-2024
==================
**Testing**  - Linq in netposition in Admine Module with PratikBhai
**Merging** - Linq in netposition in Admine Module with PratikBhai

**Comment by Kartik-RestAlgo** [comment]
24-07-2024
===================
**Developement  Done** - Orderbook, Tradebook and NetPosition forms Handel at Engine module.
Testing in Pending

**Comment by Kartik-RestAlgo** [comment]
25-07-2024
=================
**Developement Done** - Login optimization while total order 1 lac and trade 1 lac ask from engine.

**Comment by Kartik-RestAlgo** [comment]
30-07-2024
======
**Testing** - Login optimization while total order 1 lac and trade 1 lac ask from engine.
**Testing** - Orderbook, Tradebook and NetPosition forms Handel at Engine module.

**Comment by Kartik-RestAlgo** [comment]
31-07-2024
============
**Testing** - Login Optimize for more than 1 lac order and trade
**Developement Done** - NSECM tokens add in market watch.
**Developement Done** - OrderHistory send to admine and trading from engine 
**Developement Pending** - OrderHistory receive from engine to admin and trading

**Comment by Kartik-RestAlgo** [comment]
01-08-2024
===============

**Developement Done** - OrderHistory send to admine and trading from engine, Pnding 1 task
**Developement Done** - OrderHistory receive from engine to admin 
**Merging** - Token Format Change in Market Watch

**Comment by Kartik-RestAlgo** [comment]
02-08-2024
=======
**Developement Done** - UserLock not working 
**Developement Done** - Merge Market watch code to virag bhai For New Token format
**Helping** - helping Usha madem in its code.
**Testing** - Net Position in admine module in test with Pratik chauhan

**Comment by Kartik-RestAlgo** [comment]
03-08-2024
==============
**Merging** - F7 addtoscript form Merging to anilbhai
**Developement Done** - UserLock not working
**Testing** - Config save on escape

**Comment by Kartik-RestAlgo** [comment]
05-08-2024
============
**Developement Done** - order history Form will close on Esc
**Developement Done** - Login issue in second time. #256
**Developement Done** - Add Indices form in Admin Module #370
**Developement Done** - Remove "Square Up All" option from Net Position #373
**Developement In Process** - Display Logged in Admin's own limit utilized in limit tool bar
**Developement Done** - User lock not working in Trading Application

**Comment by Kartik-RestAlgo** [comment]
06-08-2024
==============
**Developement Done** - New Report Make Risk Net Position form
**Developement Done** - Changes in NetPosition in Admin and Trading

**Comment by Kartik-RestAlgo** [comment]
07-08-2024
==============
**Developement Done** - Sorting in Net Position,
changes in Net Position in admin trading and engine

**Comment by Kartik-RestAlgo** [comment]
08-08-2024
==============
**Developement Done** - Lock Screen and screen not accessible and when True password enter than only screen accessible
**Developement Pending** - Add templet in Templet Master]
**New Developement** - merge trading system from PratikSir and testing algo order and manual order and testing engine.

**Comment by Kartik-RestAlgo** [comment]
09-08-2024
===============
**Developement Done**- Add templet in Templet Master
**Developement Pending** - Sorting In net Position

**Comment by Kartik-RestAlgo** [comment]
13-08-2024
===============
Verify OrderBook, TradeBook, NetPosition, and RiskNetPosition in the Engine.
Inspect Engine Code for Trading Connection.
Ensure the Order Book and Trade Book are populated upon first login and update the Order Book data every minute.

**Comment by Kartik-RestAlgo** [comment]
14-08-2024
===========
Developement Done- Sorting In net Position
Developement Pending - Sorting In ffrmAlgoMarketWatch
testing tradebook and orderbook

**Comment by Kartik-RestAlgo** [comment]
16-08-2024
=============
Developement InProgress - New form Brokerage master.
Developement Done - 2 new table create in Database brokerage master and brokerage template.
Helping - Usha madem for refresh label in new algo market watch

**Comment by Kartik-RestAlgo** [comment]
17-08-2024
==================
Learn about concurrent Dictionary
Developement InProgress - New form Brokerage master.

**Comment by Kartik-RestAlgo** [comment]
20-08-2024
===================
**Developement InProgress** - New form Brokerage master.
**Developement InProgress**  -  Make new API for get and Post data

**Comment by Kartik-RestAlgo** [comment]
21-08-2024
==========

Developement InProgress - New form Brokerage master.
Developement InProgress - Make new API for get and Post data
Testing - Bulk order add in list and dictionary and save in .json file and .bin file

**Comment by Kartik-RestAlgo** [comment]
22-08-2024
=================

**Developement Done** - Changes in Brokerage master form.
**Developement Done** - Changes in API for get and Post data.
and Testing

**Comment by Kartik-RestAlgo** [comment]
23-08-2024
============
**Developement Done** - Brokerage master form.
**Testing and Merging**

**Comment by Kartik-RestAlgo** [comment]
24-08-2024
============
**Changes Done** - in Template Master when Leave from data grid view Value become blank in
gross exposure, buy exposure, sell exposure, margin, MTM loss, turnover
**Changes Done** - In ban script, restrict script, free script form Security type spelling is wrong.
**Changes Done** - Hide Expense template form from menu.
**Changes Done** - In strategy master form exchange spelling is wrong
**Changes Done** - Template master & limit master form ma deposit ni andar clearing ni jagya par cleaning chhe jene clearing karvu.
**Changes Done** - Change the domestic name to deposit in Template master & Limit master form.
**Changes Done** - Admin Module ni anadr aavela Start , Stop and Save button remove karva (Hide all 3 buttons).

**Comment by Kartik-RestAlgo** [comment]
29-08-2024
=============
**developement Pending** - Indicies tool bar in Trading system

**Comment by Kartik-RestAlgo** [comment]
30-08-2024
============
**Developement Done** - Change in Token seletor for Strike price remain same after change CE/PE
**Developement Done** - M2M showing Different in limit master and Net Position
**Developement Done** - Error Solving for Orderbook and Netposition given by Amanbhai

**Comment by Kartik-RestAlgo** [comment]
31-08-2024
============
**Developement Done** - Color Applying in Grid (Helping anilbhai)
**Developement Done** - in frmalgorule if change in 1st lag same value reflect in other leg.

**Comment by Kartik-RestAlgo** [comment]
02-09-02024
================
**Merging Done** - New Nt Position merge to Pratikbhai in Live Admin
**Developement Pending** - Change broadcast IP/Port dynamically and give effect while exe running

**Comment by Kartik-RestAlgo** [comment]
03-09-02024
==========
Developement Pending - Change broadcast IP/Port dynamically and give effect while exe running
Helping - Sakshi and Drashti In solving error

**Comment by Kartik-RestAlgo** [comment]
04-09-2024
===========
**Development Pending** - Implement dynamic change of broadcast IP/Port and ensure it takes effect while the executable is running. Now on Hold
**Development Done** - Merged application style configuration from Usha Madam - Pending Testing.
**Error Resolution** - Solving errors in the live trading system with Pratikbhai.

**Comment by Kartik-RestAlgo** [comment]
05-09-2024
========
**Development Pending** - Save Order and Trade inDatabase

**Comment by Kartik-RestAlgo** [comment]
06-09-2024
==========
**Development Done**- Save Order and Trade inDatabase

**Comment by Kartik-RestAlgo** [comment]
09-09-2024
==========
**Development in Process** - apply trigger on insert in currentorderhistory table
**Development in Process** - save orderbook and tradebook at day end.

**Comment by Kartik-RestAlgo** [comment]
10-09-2024
==============
**Development Done** - save TradeBook and Orderbook in database while engine close
**Development Done** - new field add NetAmount Helping sakshi and merger to pratikbhai

**Comment by Kartik-RestAlgo** [comment]
11-09-2024
==========
**Development in Process** - save TradeBook and Orderbook in database in Binary file
Testing Bulk order for save and Load

**Comment by Kartik-RestAlgo** [comment]
12-09-2024
========
**Development in Process** - save TradeBook and Orderbook in database in Binary file
Testing Bulk order for save and Load

**Comment by Kartik-RestAlgo** [comment]
16-09-2024
======
Merging NetPosition in Trading system From Module

**Comment by Kartik-RestAlgo** [comment]
17-09-2024
================
Merging NetPosition in Trading system From Module
Changing NetPosition in Admin Module
Testing Done by KAR/PPB/Dipakbhai.

**Comment by Kartik-RestAlgo** [comment]
18-09-2024
=========
**Testing With PratikBhai** - Save and load OrderBook,TradeBook And NetPosition
**Developement Done** - Set I-Set same as G-Set if I-Set is Null
**Developement Done** - Design For Individual parameter set In all I-Set same As G-Set and add Extra button for save G-Set to all I-Set

**Comment by Kartik-RestAlgo** [comment]
19-09-2024
===============
**Developement done** - Set Default OrderLot in FOrderLot and ROrderLot in Algo rule
**Testing** - Individual parameter set In all I-Set same As G-Set and add Extra button for save G-Set to all I-Set

**Comment by Kartik-RestAlgo** [comment]
20-09-2024
========
**Developement done** - Set Default OrderLot in FOrderLot and ROrderLot in Algo rule
**Developement done** - Individual parameter set In all I-Set same As G-Set and add Extra button for save G-Set to all I-Set
Testing With PratikBhai - For Extra Trade Received

**Comment by Kartik-RestAlgo** [comment]
21-09-2024
===========
**Developement done** - save last location of indexViewForm
Helping @Sakshi-Restalgo 
Helping @drashti-rydt to Complete its Task.

**Comment by Kartik-RestAlgo** [comment]
23-09-2024
===========
Merging 2 task to Pratik Bhai
Merging 2 Task to Virag Bhai

**Comment by Kartik-RestAlgo** [comment]
24-09-2024
=============
Testing with support team
Solving Errors of TokenSelector and G-Set/I-Set

**Comment by Kartik-RestAlgo** [comment]
25-09-2024
===========
Solving Errors of TokenSelector and G-Set/I-Set
Solving Errors of NetPosition

**Comment by Kartik-RestAlgo** [comment]
26-09-2024
==========
Working on Token Selector Errors and Learning about python and C#

**Comment by Kartik-RestAlgo** [comment]
27-09-2024
===
Developing Demo of Datamodel vs ViewModel

**Comment by Kartik-RestAlgo** [comment]
28-09-2024
=
**Developing Done** - Change Token Selector
Demo To support for New Exe Log Manager and Engine exe

**Comment by Kartik-RestAlgo** [comment]
30-09-2024
========
Merging Userwise Common Config Class into trading system and in Admin Module

**Comment by Kartik-RestAlgo** [comment]
03-10-2024
=
Merging MISC Changes to Dipakbhai
and Helping Sakshi and Drashti For Solving errors
And Merging New Token Selector to Aman Code

**Comment by Kartik-RestAlgo** [comment]
04-10-2024
=====
Working With PratikBhai For Template save in Limit Details
Working With Sakshi and Drashti For Merging New Net Position.

**Comment by Kartik-RestAlgo** [comment]
05-10-2024
=
Working With PratikBhai For Engine Merging Net Position in Engine.
Working With Sakshi and Ushaben For Tokenselector, frmAddtoScripts

**Comment by Kartik-RestAlgo** [comment]
09-10-2024
======
Working on Tokenselector for Errors,
Working With Drashti ben and Usha ben For Tokenselector, frmAddtoScripts

**Comment by Kartik-RestAlgo** [comment]
14-10-2024
=======
Merging Common Config Class To Viragbhai and NXBlock

**Comment by Kartik-RestAlgo** [comment]
15-10-2024
=====
Change in Token Selectors.

**Comment by Kartik-RestAlgo** [comment]
16-10-2024
=======
Developement InProcess - Creating Common Column Profile if Column not exist of any Class

**Comment by Kartik-RestAlgo** [comment]
17-10-2024
======
Developement Done - Creating Common Column Profile if Column not exist of any Class
Trying To Change in Common for saving Column Profile

**Comment by Kartik-RestAlgo** [comment]
18-10-2024
==========
Merging Common Config Class to AnilBhai in Trading System of Shiddarth Client.

**Comment by Kartik-RestAlgo** [comment]
22-10-2024
========
Solving Error of Orderbook and net position while engine restart
solving error while open normal buy/Sell form from Market Depth
Learning about .Net core and .Net

**Comment by Kartik-RestAlgo** [comment]
24-10-2024
======
Solving Bug of Duplicate trade
Solving Error of Log Manager

**Comment by Kartik-RestAlgo** [comment]
25-10-2024
========
Change Logic of Log Manager for saving NetPosition 
orderbookHistory, tradebook
